devito/tools/utils.py

```diff
@@ -356,3 +357,29 @@ def transitive_closure(R):
         visited = []
         ans[k] = reachable_items(R, k, visited)
     return ans
+
+
+def humanbytes(B):
+    """
+    Return the given bytes as a human friendly KB, MB, GB, or TB string.
+
+    Extracted and then readapted from:
+        https://stackoverflow.com/questions/12523586/python-format-size-\
+                application-converting-b-to-kb-mb-gb-tb
+    """
+    B = float(B)
+    KB = float(1024)
+    MB = float(KB ** 2)  # 1,048,576
+    GB = float(KB ** 3)  # 1,073,741,824
+    TB = float(KB ** 4)  # 1,099,511,627,776
+
+    if B < KB:
+        return '%d %s' % (int(B), 'B')
+    elif KB <= B < MB:
+        return '%d KB' % round(B / KB)
+    elif MB <= B < GB:
+        return '%d MB' % round(B / MB)
+    elif GB <= B < TB:
+        return '%d GB' % round(B / GB)
+    elif TB <= B:
+        return '%d TB' % round(B / TB)
```

Comment:
So we only care about "closest integer" value? You can also reverse the order of the if to simplify the conditions.


Slightly more compact version too

```
for unit in ['B', 'KB', 'MB', 'GB', 'TB']:
        if B < 1024:
            break
        B /= 1024
    return '%d %s' % (round(B), unit)
```
```

Reply:
yup, I copy-pasted from stackoverflow

Reply:
ok